src/DataStructures/VectorImpl.hpp

```diff
@@ -297,43 +297,56 @@ class VectorImpl
 
  protected:
   std::unique_ptr<value_type[]> owned_data_{};
+  std::array<T, StaticSize> static_owned_data_{};
   bool owning_{true};
 
+  // This should only be called if we are owning. If we are not owning, then
+  // neither owned_data_ or static_owned_data_ actually has the data we want.
   SPECTRE_ALWAYS_INLINE void reset_pointer_vector(const size_t set_size) {
     if (set_size == 0) {
       return;
     }
-    if (owned_data_ == nullptr) {
+    if (owned_data_ == nullptr and set_size > StaticSize) {
       ERROR(
           "VectorImpl::reset_pointer_vector cannot be called when owned_data_ "
           "is nullptr.");
     }
-    this->reset(owned_data_.get(), set_size);
+
+    if (set_size <= StaticSize) {
+      this->reset(static_owned_data_.data(), set_size);
+      // Free memory if downsizing
+      owned_data_ = nullptr;
+    } else {
+      this->reset(owned_data_.get(), set_size);
+    }
+  }
+
+  SPECTRE_ALWAYS_INLINE std::unique_ptr<value_type[]> heap_alloc_if_necessary(
+      const size_t set_size) {
+    return set_size > StaticSize
+               ? cpp20::make_unique_for_overwrite<value_type[]>(set_size)
+               : nullptr;
   }
 };
 
-template <typename T, typename VectorType>
-VectorImpl<T, VectorType>::VectorImpl(const VectorImpl<T, VectorType>& rhs)
-    : BaseType{rhs},
-      owned_data_(
-          rhs.size() > 0
-              ? cpp20::make_unique_for_overwrite<value_type[]>(rhs.size())
-              : nullptr) {
+/// \cond HIDDEN_SYMBOLS
```

Comment:
Why hide these?

Reply:
I was getting errors when building the docs that Doxygen wasn't able to find some symbols. The declarations are already hidden above, so I just hid the definitions as well.